.github/workflows/pipeline.yml

```diff
@@ -6,19 +6,27 @@ on:
     branches-ignore:
       - 'main'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
This setting will cause the pipeline to abort any existing pipeline runs for this branch when we push another change to this branch. This will just help us waiting forever and wasting so many resources when we're repeatedly committing and pushing quickly